src/cpp/elf/elfwriter.cpp

```diff
@@ -219,6 +219,29 @@ add_text_data_section(const std::vector<writer>& mwriter, std::vector<symbol>& s
   }
 }
 
+void
+elf_writer::
+add_symtab(const std::string& name, const std::string& /*dec*/)
+{
+
+  ELFIO::section* str_sec = m_elfio.sections.add( ".strtab" );
```

Comment:
Every time `add_symtab()` is called, we are creating `.strtab` and `.symtab` sections which looks incorrect. These sections should be created once by the `elf_writer` may be in its constructor. `add_symtab()` should just add the requested symbol.